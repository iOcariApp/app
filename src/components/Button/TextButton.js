import React from "react";
import { PropTypes } from "prop-types";

import Button from "./Button";

const TextButton = ({ color, ...rest }) => (
  <Button color={color} backgroundColor="transparent" {...rest} />
);

TextButton.propTypes = {
  color: PropTypes.string,
};
TextButton.defaultProps = {
  color: "white",
};

export default TextButton;
