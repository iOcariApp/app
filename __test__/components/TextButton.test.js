import React from "react";
import renderer from "react-test-renderer";

import TextButton from "components/Button/TextButton";

test("renders correctly", () => {
  const tree = renderer.create(<TextButton />).toJSON();
  expect(tree).toMatchSnapshot();
});

test("applies props correctly", () => {
  const tree = renderer.create(<TextButton color="yellow" />).toJSON();
  expect(tree).toMatchSnapshot();
});
