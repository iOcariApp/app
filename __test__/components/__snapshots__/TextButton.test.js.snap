// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applies props correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      Array [
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "width": "100%",
        },
        Object {},
      ],
      Object {
        "borderRadius": 4,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderRadius": 4,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "height": 56,
            },
            Object {},
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "yellow",
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 18,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "marginLeft": 15,
        "marginRight": 15,
      },
      undefined,
      Array [
        Object {
          "marginBottom": 0,
          "marginLeft": 0,
          "marginRight": 0,
          "marginTop": 0,
          "width": "100%",
        },
        Object {},
      ],
      Object {
        "borderRadius": 4,
      },
    ]
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      pointerEvents="box-only"
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9E9E9E",
            "flexDirection": "row",
            "justifyContent": "center",
            "padding": 19,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderRadius": 4,
          },
          Object {
            "padding": 12,
          },
          undefined,
          undefined,
          undefined,
          Array [
            Object {
              "height": 56,
            },
            Object {},
          ],
          undefined,
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "white",
                "fontSize": 20,
              },
              Object {
                "color": "white",
              },
              Object {
                "fontSize": 17.5,
              },
              Object {
                "fontSize": 18,
              },
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          ]
        }
      />
    </View>
  </View>
</View>
`;
